MovieApp/AppRouter.swift

```diff
@@ -0,0 +1,42 @@
+
+import Foundation
+import UIKit
+import MovieAppData
+
+class AppRouter {
+    
+    private let navigationController: UINavigationController!
+    
+    init(navigationController: UINavigationController) {
+        self.navigationController = navigationController
+    }
+    
+    func setStartScreen(in window: UIWindow?) {
```

Comment:
Although you have achieved the correct behavior, it is not implemented in the correct way. What you did is:
* You have root NavigationController, passed in the constructor
* You have pushed MovieViewController in the root NavigationController
* You have assigned root NavigationController and FavoritesVC in the TabVC

The correct way would be:
* In the constructor, initialize two additional NavigationControllers and assign them to a class field
* One NC is for the Movie tab, other one is for the Favorites tab
* Push MovieVC to the first NC and push FavoritesVC to the second NC
* Assign those two NC to the TabVC
* In `showMovie`, use movie NC to push `MovieDetailsVC`